src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -188,12 +195,21 @@ public void testDeleteDomainField() throws Exception
         domainFormPanel.getField("deleteMe")
                 .clickRemoveField()
                 .dismiss("Yes");
-        domainDesignerPage.clickSaveChanges();
+        domainDesignerPage.clickSaveAndFinish();
 
         GetDomainCommand domainCommand = new GetDomainCommand("exp.materials", sampleSet);
         DomainResponse afterResponse = domainCommand.execute(createDefaultConnection(true), getProjectName());
         List<Map<String, Object>> remainingFields = afterResponse.getColumns();
         assertEquals("expect only field in the domain to have been deleted", 0, remainingFields.size());
+
+        // double-check to ensure the column has been deleted
+        SelectRowsResponse rowsResponse = dgen.getRowsFromServer(createDefaultConnection(true));
+        // this column should no longer exist
+        List<Map<String, Object>> deleteMe = rowsResponse.getColumnModel().stream().filter(a -> a.get("dataIndex").equals("deleteMe")).collect(Collectors.toList());
+        assertEquals(0, deleteMe.size());
+        // make sure the name field is still there
+        List<Map<String, Object>> name = rowsResponse.getColumnModel().stream().filter(a -> a.get("dataIndex").equals("Name")).collect(Collectors.toList());
+        assertEquals(1, name.size());
```

Comment:
Collect the list of columns so that assertion error messages can be more informative.
```
SelectRowsResponse rowsResponse = dgen.getRowsFromServer(createDefaultConnection(true));
List<String> columnsAfterDelete = rowsResponse.getColumnModel().stream().map(col -> (String) col.get("dataIndex")).collect(Collectors.toList());
Assert.assertThat("Columns after delete", columnsAfterDelete, CoreMatchers.allOf(CoreMatchers.hasItem("Name"), CoreMatchers.not(CoreMatchers.hasItem("deleteMe"))));
```

Reply:
nice